php/libraries/ICandidate.class.inc

```diff
@@ -0,0 +1,319 @@
+     * Returns the list of visit labels w/ corresponding SessionID as key
+     *
+     * @return array<int,string> Associative array in the format
+     *                              SessionID=>Visit_label
+     */
+    public function getListOfVisitLabels(): array;
+
+    /**
+     * Generates list of valid subprojects for a the selected Candidate's
+     * project
+     *
+     * @return array<int,int> Array of the form SubprojectID => SubprojectID
+     *
+     * @throws DatabaseException
+     */
+    public function getValidSubprojects(): array;
+
+    /**
+     * Returns the Visit_label for a subject's first TimePoint.
+     *
+     * @return string The visit label for the candidate's first visit.
+     *                  Returns an empty string if one doesn't exist.
+     */
+    public function getFirstVisit(): string;
+
+    /**
+     * Returns the next visit number that should be created for this
+     * candidate.
+     *
+     * @return int The next visit number
+     */
+    public function getNextVisitNo() : int;
+
+    /**
+     * Returns list of consents and their respective statuses for this candidate
+     *
+     * @return array List of consents and their associated values for this candidate
+     *               The keys of the arrays are the IDs of the consents
+     */
+    public function getConsents(): array;
+
+    /**
+     * Gets the participant_status options from participant_status_options
+     * table. Used to populate LorisForm.
+     *
+     * @return string[]
+     */
+    static function getParticipantStatusOptions(): array;
+
+    /**
+     * Gets the participant_status options suboptions from participant_status_options
+     *
+     * @param int $parentID parent ID of the participant status option
+     *
+     * @return string[] Options array suitable for use in LorisForm select
+     *               element
+     */
+    public static function getParticipantStatusSubOptions(int $parentID): array;
+
+    /**
+     * Sets generic data in this class to $newData.
+     * Manages the parameter_candidate table so that projects have a way to add
+     * whatever data they want which is outside the scope of LORIS to their
+     * candidates.
+     *
+     * @param array $newData Key-value pair(s) of the field(s) to change and the
+     *                              new value(s).
+     *
+     * @return void
+     */
+    public function setData(array $newData): void;
+
+    /**
+     * Validate's that a PSCID matches the format defined by project
+     * config
+     *
+     * @param string $PSCID The PSCID to validate
+     *
+     * @return bool
+     */
+    public static function validatePSCID(string $PSCID): bool;
+
+    /**
+     * This is a function to supply data about the Candidate object to the
+     * front-end. Many Smarty templates refer to the specific column names
+     * and aliases used below.
+     * This function exists to provide legacy support. It should not be used
+     * by new code.
```

Comment:
If it only exists to provide legacy support, I don't see why it should be in the new interface